package Cmd;

enum Command
{
  LOGIN_USER_PROTOCMD = 1;
  ERROR_USER_PROTOCMD = 2;

  SCENE_USER_PROTOCMD = 5;
  SCENE_USER_ITEM_PROTOCMD = 6;
  SCENE_USER_SKILL_PROTOCMD = 7;
  SCENE_USER_QUEST_PROTOCMD = 8;
  SCENE_USER2_PROTOCMD = 9;
  SCENE_USER_PET_PROTOCMD = 10;
  FUBEN_PROTOCMD = 11;
  SCENE_USER_MAP_PROTOCMD = 12;
  SCENE_USER_MOUNT_PROTOCMD = 13;

  SCENE_BOSS_PROTOCMD = 15;
  SCENE_USER_CARRIER_PROTOCMD = 16;
  SCENE_USER_ACHIEVE_PROTOCMD = 17;
  SCENE_USER_TIP_PROTOCMD = 18;
  SCENE_USER_CHATROOM_PROTOCMD = 19;
  INFINITE_TOWER_PROTOCMD = 20;
  SCENE_USER_SEAL_PROTOCMD = 21;
  SCENE_USER_INTER_PROTOCMD = 22;
  SCENE_USER_MANUAL_PROTOCMD = 23;
  SCENE_USER_CHAT_PROTOCMD = 24;
  USER_EVENT_PROTOCMD = 25;
  SCENE_USER_TRADE_PROTOCMD = 26;
  SCENE_USER_AUGURY_PROTOCMD = 27;
  SCENE_USER_ASTROLABE_PROTOCMD = 28;
  SCENE_USER_FOOD_PROTOCMD = 29;
  SCENE_USER_PHOTO_PROTOCMD = 30;
  SCENE_USER_TUTOR_PROTOCMD = 31;
  SCENE_USER_BEING_PROTOCMD = 32;

  SESSION_USER_GUILD_PROTOCMD = 50;
  SESSION_USER_TEAM_PROTOCMD = 51;
  SESSION_USER_SHOP_PROTOCMD = 52;
  SESSION_USER_WEATHER_PROTOCMD = 53;
  SESSION_USER_MAIL_PROTOCMD = 55;
  SESSION_USER_SOCIALITY_PROTOCMD = 56;

  RECORD_USER_TRADE_PROTOCMD = 57;
  DOJO_PROTOCMD = 58;
  CHAT_PROTOCMD = 59;
  ACTIVITY_PROTOCMD = 60;
  MATCHC_PROTOCMD = 61;
  SESSION_USER_AUTHORIZE_PROTOCMD = 62;
  AUCTIONC_PROTOCMD = 63;
  ACTIVITY_EVENT_PROTOCMD = 64;
  WEDDINGC_PROTOCMD = 65;

  PVE_CARD_PROTOCMD = 66;

  TEAM_RAID_PROTOCMD = 67;

  SESSION_OVERSEAS_TW_PROTOCMD = 80;

  CLIENT_CMD = 99;
  MAX_USER_CMD = 100;

  RECORD_DATA_PROTOCMD = 200;
  TRADE_PROTOCMD = 201;
  SESSION_PROTOCMD = 202;
  GMTOOLS_PROTOCMD = 203;
  LOG_PROTOCMD = 204;
  GATE_SUPER_PROTOCMD = 205;
  REGION_PROTOCMD = 206;
  STAT_PROTOCMD = 207;
  SOCIAL_PROTOCMD = 208;
  TEAM_PROTOCMD = 209;
  GUILD_PROTOCMD = 210;
  GZONE_PROTOCMD = 211;
  MATCHS_PROTOCMD = 212;
  AUCTIONS_PROTOCMD = 213;
  WEDDINGS_PROTOCMD = 214;
  BOSSS_PROTOCMD = 216;

  REG_CMD = 253;
  GATEWAY_CMD = 250;

  SYSTEM_PROTOCMD = 255;
}

message Nonce
{
  optional uint32 timestamp = 1;
  optional uint32 index = 2;
  optional string sign = 3;
}
